duckduckgo/macos-browser
# Add Phishing Detection Integration [Main PR]

Task/Issue URL: https://app.asana.com/0/0/1207334681074772/f
Tech Design URL: https://app.asana.com/0/481882893211075/1207156899292810/f
CC: https://app.asana.com/0/481882893211075/1207220724600204/f

**Description**:
Implement macOS phishing detection integration to facilitate end-to-end phishing error pages. This includes: embedded datasets, data updating, data storage, error page, tab extension, privacy dashboard, preferences page, and remote config.

**Steps to test this PR**:
1. Build on macOS
2. Ensure signed in via use-login.duckduckgo.com (only available internally)
3. Quit the app
4. Visit https://privacy-test-pages.site/security/badware/phishing.html
5. Ensure error page is thrown
6. Click advanced
7. Click "Accept Risk"
8. Ensure page loads
9. Play around with navigations (back/forward, etc.)
10. Disable the feature in Settings>General
11. Try edge case test pages:
 - https://bad.third-party.site/security/badware/phishing-iframe-loader.html
 - https://bad.third-party.site/security/badware/phishing-meta-redirect.html - failing
 - https://bad.third-party.site/security/badware/phishing-js-redirector-helper.html - failing

The last two failing test cases are to be addressed in follow-up work. Currently this feature is hidden behind a feature-flag for internal-only builds so we've considered the risk low.



**Definition of Done**:


---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
